app/Models/Store/Order.php

```diff
@@ -82,6 +82,17 @@ public function scopeWhereOrderNumber($query, $orderNumber)
         ]);
     }
 
+    public static function scopeWherePaymentTransactionId($query, $transactionId)
+    {
+        $ordersTable = (new static)->getTable();
+        $paymentsTable = (new Payment)->getTable();
+
+        return $query
+            ->join($paymentsTable, "{$paymentsTable}.order_id", '=', "{$ordersTable}.order_id")
```

Comment:
`whereIn('order_id', Payment::select(...)->where()...)` for less table name hacks?